code/modules/projectiles/projectile/beams.dm

```diff
@@ -290,26 +291,38 @@ var/list/beam_master = list()
 		kill_count--
 		bump_original_check()
 
+		var/AX = (starting.x - src.x)*32

```

Comment:
But it can be written in such a way as to remove all duplicated code without much effort